Catch mail exceptions

dariomory opened the pull request:
Fixes #403

_______

**Changes**

- Added different behavior admin and cms invitations

joeribekker reviewed (changes requested):
Please keep this more simple by adding the appropriate message in the part of the code where the functional action takes place.

Now you have messages in the `InviteUser` class that differ because they functionally do different things but this is done outside of the `InviteUser` class and thus is very confusing.

KIS = Keep It Simple :)

dariomory commented:
@joeribekker Your point about the KIS principle is good. I have been thinking carefully about the simplest solution that sticks to both dry and solid principles.

A few changes:

- Simplified to prevent confusion. The custom message callable is just an alternative to a regular event function.
- Added two tests (for admin and CMS)

joeribekker commented:
I find the solution far too difficult for what it is, so I'm closing this for now since it works just fine if the configuration is valid.
